File: src/LIT.Smabu.UseCases/Offers/OfferItemDTO.cs
using LIT.Smabu.Domain.CatalogAggregate;
using LIT.Smabu.Domain.Common;
using LIT.Smabu.Domain.OfferAggregate;
using LIT.Smabu.UseCases.Base;
using System.Globalization;

namespace LIT.Smabu.UseCases.Offers
{
    public record OfferItemDTO(OfferItemId Id, OfferId OfferId, string DisplayName, int Position, string Details, Quantity Quantity, decimal UnitPrice, decimal TotalPrice, CatalogItemId? CatalogItemId) : IDTO
    {
        public static OfferItemDTO Create(OfferItem offerItem)
        {
            return new(offerItem.Id, offerItem.OfferId, offerItem.DisplayName, offerItem.Position, offerItem.Details,
                offerItem.Quantity, offerItem.UnitPrice, offerItem.TotalPrice, offerItem.CatalogItemId);
        }
    }
}

File: src/lit.smabu.react/src/types/domain/offer-item-dto.ts
import { CatalogItemId } from './catalog-item-id';
import { OfferId } from './offer-id';
import { OfferItemId } from './offer-item-id';
import { Quantity } from './quantity';

export interface OfferItemDTO {
  id?: OfferItemId;
  offerId?: OfferId;
  position?: number;
  details?: string | null;
  quantity?: Quantity;
  unitPrice?: number;
  totalPrice?: number;
  catalogItemId?: CatalogItemId;
  displayName?: string | null;
}
